engine/src/main/java/org/hibernate/search/engine/impl/WorkPlan.java

```diff
@@ -145,7 +147,8 @@ public void processContainedInAndPrepareExecution() {
 	 */
 	public <T> void recurseContainedIn(T value, ContainedInRecursionContext context, String tenantId) {
 		Class<T> entityClass = instanceInitializer.getClass( value );
-		PerClassWork classWork = getClassWork( tenantId, entityClass );
+		//TODO separate the ContainedIn processing in its own registry of types
```

Comment:
You can't forget about it. If you ever try to figure out how to encode non-pojo types you'll have to rewrite this... it's useful to have a TODO here to help with "what do I do here" at that point.